common/achievement/Readme.md

```diff
@@ -1,4 +1,352 @@
 # Achievement
 
-Achievements are a part of the Gamification.
-Documentation follows...
+Achievements are a part of the Gamification. They are stored in the database as user_achievement and should not be mistaken with the achievement_template database table. The distinct differences are explained in Chapter [1.1 Database Models](#database-models)
+
+### Table of Contents
+
+1.  [Features](#features)
+    1. [Database Models](#database-models)
+    2. [Components](#components)
+    3. [Resolvers](#resolvers)
+2.  [Achievement System](#achievement-system)
+    1. [Reward Action](#reward-action)
+    2. [Get or Create User Achievement](#get-or-create-user-achievement)
+    3. [Create Achievement](#create-achievement)
+    4. [Check User Achievement](#check-user-achievement)
+    5. [Evaluate Achievement](#evaluate-achievement)
+3.  [Buckets](#buckets)
+    1.  [Bucket Concept](#bucket-concept)
+    2.  [Bucket Types](#bucket-types)
+    3.  [Aggregation Functions](#aggregation-functions)
+    4.  [Aggregators](#aggregators)
+    5.  [Data Aggregation](#data-aggregation)
+4.  [Conditions](#conditions)
+    1.  [Used Technologies](#used-technologies)
+    2.  [Writing Conditions](#writing-conditions)
+5.  [Metrics](#metrics)
+6.  [Tracking Events](#tracking-events)
+
+## 1. Features
+
+-   Create Achievement Templates: Administrators can create new achievement templates inside the database
+-   Create Achievements: New achievements get created when related events are triggered by a users action or when previous achievements of the same group are marked as achieved
+-   Evaluate Achievements: Achievements are evaluated for fulfillment of the conditions to either issue the reward or update a started achievement.
+
+### 1.1. Database Models
+
+| achievement_template                                                                  | user_achievement                                                                                                      |
+| :------------------------------------------------------------------------------------ | :-------------------------------------------------------------------------------------------------------------------- |
+| predefined blueprint in the database that is used to create new achievements from it. | achievement data related to a user that stores information about the achievement, its status and when it was achieved |
+
+#### achievement_template
+
+| field                                                               | description                                                                                                                                                             |
```

Comment:
I like the idea of these tables, but I don't think they are adding a lot of value here.
I would suggest to remove them from this doc, but move the description to the `schema.prisma`.

Reply:
Maybe it makes sense to replace the tables with a high level ER diagram. It doesn't need to be complete as it will be outdated over time anyways.